moveit_core/transforms/include/moveit/transforms/transforms.h

```diff
@@ -51,8 +52,8 @@ MOVEIT_CLASS_FORWARD(Transforms);
 
 /// @brief Map frame names to the transformation matrix that can transform objects from the frame name to the planning
 /// frame
-typedef std::map<std::string, Eigen::Isometry3d, std::less<std::string>,
-                 Eigen::aligned_allocator<std::pair<const std::string, Eigen::Isometry3d> > >
+typedef std::map<std::string, Eigen::Affine3d, std::less<std::string>,
+                 Eigen::aligned_allocator<std::pair<const std::string, Eigen::Affine3d> > >
```

Comment:
This was a big change we made to moveit in Dec, why is it being reverted here?

https://github.com/ros-planning/moveit/pull/1096

Reply:
Partially described at https://github.com/AcutronicRobotics/moveit2-original/pull/9. See https://github.com/ros2/geometry2/blob/ros2/tf2_eigen/include/tf2_eigen/tf2_eigen.h#L46.

Reply:
Going back to Affine (or cluttering the history with hundreds of lines of unnecessary revert/counter-revert) is not an option. If you found a problem with ros2's geometry2-package, this should be fixed there (and this repository should probably require the patched version until then.)

I would hope upstream is very open to fixes in their interfaces if we need them.

Reply:
agreed, we definitely should not revert back to Affine3d, that's a performance hit

i just looked into the difference between melodic-devel and ros2 branch and i don't see any changes in functions for in [tf2_eigen.h](https://github.com/ros/geometry2/compare/melodic-devel...ros2:ros2). why does this require changing for moveit2?

Reply:
@davetcoleman and @v4hn, have a look at the ROS 2 geometry2 package as indicated above. For further details:
- ROS1: https://github.com/ros/geometry2/blob/melodic-devel/tf2_eigen/include/tf2_eigen/tf2_eigen.h
- ROS2: https://github.com/ros2/geometry2/blob/ros2/tf2_eigen/include/tf2_eigen/tf2_eigen.h#L46

There's been a switch to `Affine3d`. Maybe @clalancette could advice if they're willing to accept a change reverting to `Isometry3d`? We should try to reach a compromise before producing any further changes.

Reply:
@vmayoral: the switch was actually *to* `Isometry3D` (as that makes much more sense). There's nothing to revert.

See:

 - https://github.com/ros-planning/geometric_shapes/pull/88
 - https://github.com/ros-planning/moveit/pull/1096

It seems `ros2/geometry2` just hasn't been kept up-to-date (example of ROS1 - ROS2 community/maintenance split?).

Reply:
Thanks @gavanderhoorn for clarifying.

> It seems `ros2/geometry2` just hasn't been kept up-to-date (example of ROS1 - ROS2 community/maintenance split?).

I guess so. 

@clalancette, how shall we proceed about this? Shall I book some time next week and provide a PR to adjust things in geometry2 for ROS2 or do you want to take care of this yourself? We can then re-adjust things here afterwards.

Reply:
This is the PR that needs to be forward ported: https://github.com/ros/geometry2/pull/335

Reply:
See https://github.com/ros-planning/moveit2/pull/12/commits/94ebbcc191db16d6e3fae760d2d8b8062517d53f. I believe we should be fine with this, right @davetcoleman?